src/main/java/main/movie/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package main.movie.service.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import main.movie.dao.ShoppingCartDao;
+import main.movie.dao.TicketDao;
+import main.movie.lib.Inject;
+import main.movie.lib.Service;
+import main.movie.model.MovieSession;
+import main.movie.model.ShoppingCart;
+import main.movie.model.Ticket;
+import main.movie.model.User;
+import main.movie.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticket = ticketDao.create(ticket);
+        ShoppingCart shoppingCart = getByUser(user);
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user);
+    }
+
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCart.setTickets(new ArrayList<>(Collections.emptyList()));
```

Comment:
you can omit this - try to debug and see if null is returned when you look at `tickets` list after you call `ShoppingCart shoppingCart = getByUser(user);` in `addSession()`

also, Collections.emptyList() returns immutable list that can not be changed